contrib/platform/src/com/sun/jna/platform/mac/CoreFoundation.java

```diff
@@ -0,0 +1,573 @@
+     *
+     * @param cf
+     *            The {@code CFType} object to examine.
+     * @return A number representing the reference count of {code cf}.
+     */
+    long CFGetRetainCount(CFTypeRef cf);
+
+    /**
+     * Returns the value associated with a given key.
+     *
+     * @param theDict
+     *            The dictionary to examine.
+     * @param key
+     *            The key for which to find a match in {@code theDict}. The key hash
+     *            and equal callbacks provided when the dictionary was created are
+     *            used to compare. If the hash callback was {@code null}, the key is
+     *            treated as a pointer and converted to an integer. If the equal
+     *            callback was {@code null}, pointer equality (in C, ==) is used. If
+     *            {@code key}, or any of the keys in {@code theDict}, is not
+     *            understood by the equal callback, the behavior is undefined.
+     * @return The value associated with key in {@code theDict}, or {@code null} if
+     *         no key-value pair matching key exists. Since {@code null} is also a
+     *         valid value in some dictionaries, use
+     *         {@link #CFDictionaryGetValueIfPresent} to distinguish between a value
+     *         that is not found, and a {@code null} value.
+     */
+    CFTypeRef CFDictionaryGetValue(CFTypeRef theDict, CFStringRef key);
+
+    /**
+     * Returns a boolean value that indicates whether a given value for a given key
+     * is in a dictionary, and returns that value indirectly if it exists.
+     *
+     * @param theDict
+     *            The dictionary to examine.
+     * @param key
+     *            The key for which to find a match in {@code theDict}. The key hash
+     *            and equal callbacks provided when the dictionary was created are
+     *            used to compare. If the hash callback was {@code null}, the key is
+     *            treated as a pointer and converted to an integer. If the equal
+     *            callback was {@code null}, pointer equality (in C, ==) is used. If
+     *            {@code key}, or any of the keys in {@code theDict}, is not
+     *            understood by the equal callback, the behavior is undefined.
+     * @param value
+     *            A pointer to memory which, on return, is filled with the
+     *            pointer-sized value if a matching key is found. If no key match is
+     *            found, the contents of the storage pointed to by this parameter
+     *            are undefined. This value may be {@code null}, in which case the
+     *            value from the dictionary is not returned (but the return value of
+     *            this function still indicates whether or not the key-value pair
+     *            was present).
+     * @return {@code true} if a matching key was found, otherwise {@code false}.
+     */
+    boolean CFDictionaryGetValueIfPresent(CFDictionaryRef theDict, CFStringRef key, PointerType value);
```

Comment:
Shouldn't the `value` be a `PointerByReference`?

Reply:
This is a similar discussion to the earlier casting discussions.  You're probably right that `PointerByReference` is correct here (and probably for the return type of the `CFDictionaryGetValue()` mapping as well) and we should just require the user to cast the return value.